sdk/src/wallet/operations/syncing/mod.rs

```diff
@@ -168,102 +165,98 @@ where
-                for (account_or_nft_address, output_address) in &new_account_and_nft_addresses {
-                    let output_ids = self
-                        .get_output_ids_for_address(
-                            &Bech32Address::new(bech32_hrp, account_or_nft_address.clone()),
-                            options,
-                        )
-                        .await?;
-
-                    // Update address with unspent outputs
-                    let address_with_unspent_outputs = addresses_with_unspent_outputs
-                        .iter_mut()
-                        .find(|address| address.address.inner() == output_address)
-                        .ok_or_else(|| {
-                            crate::wallet::Error::WalletAddressMismatch(output_address.clone().to_bech32(bech32_hrp))
-                        })?;
-                    address_with_unspent_outputs.output_ids.extend(output_ids.clone());
-
-                    let new_outputs_data_inner = self.get_outputs(output_ids).await?;
-
-                    let outputs_data_inner = self
-                        .output_response_to_output_data(new_outputs_data_inner, &address_with_unspent_outputs)
-                        .await?;
-
-                    outputs_data.extend(outputs_data_inner.clone());
-                    new_outputs_data.extend(outputs_data_inner);
-                }
-                new_outputs_data
-            };
-
-            // Clear, so we only get new addresses
-            new_account_and_nft_addresses.clear();
-
-            // Add new account and nft addresses
-            for output_data in new_outputs_data {
-                match output_data.output {
-                    Output::Account(account_output) => {
-                        let account_address =
-                            AccountAddress::from(account_output.account_id_non_null(&output_data.output_id));
+        // Get the unspent and spent/not-synced output ids per address to sync
+        let (addresses_to_sync_with_unspent_output_ids, mut spent_or_not_synced_output_ids) = self
+            .get_output_ids_for_addresses(addresses_to_sync.clone(), options)
+            .await?;
 
-                        new_account_and_nft_addresses.insert(Address::Account(account_address), output_data.address);
-                    }
-                    Output::Nft(nft_output) => {
-                        let nft_address = NftAddress::from(nft_output.nft_id_non_null(&output_data.output_id));
+        // Get the corresponding unspent output data
+        let mut new_outputs_data = self
+            .get_outputs_from_address_output_ids(addresses_to_sync_with_unspent_output_ids)
+            .await?;
 
-                        new_account_and_nft_addresses.insert(Address::Nft(nft_address), output_data.address);
+        loop {
+            // Try to discover new addresses
+            // See https://github.com/rust-lang/rust-clippy/issues/8539 regarding this lint.
```

Comment:
I feel like there's a better way 🤔

Reply:
probably ... any ideas?

Reply:
I'll fiddle with it locally and get back to you